src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -301,6 +301,14 @@ public void run()
 
             final StatusResponseHolder response = client.go(request, new StatusResponseHandler(Charsets.UTF_8)).get();
 
+            if (response.getStatus().getCode() == 413) {
+              throw new ISE(
+                  "Batch size was not accepted by server at urlString[%s]! Bad maxBatchSize of %,d",
```

Comment:
We should be more specific about the error code received, since it may not be obvious to someone reading the logs, e.g. `Received HTTP status 413. Batch size may be too large, try adjusting com.metamx.emitter.http.maxBatchSize`